docker/catalog-auth-s3/docker-compose.yml

```diff
@@ -0,0 +1,176 @@
+#
+# Copyright (C) 2024 Dremio
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+# http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# To request an access token, use the following command:
+# token=$(curl http://127.0.0.1:8080/realms/iceberg/protocol/openid-connect/token --user client1:s3cr3t -d 'grant_type=client_credentials' -d 'scope=profile' | jq -r .access_token)
+
+# Then use the access token in the Authorization header:
+# curl http://127.0.0.1:19120/api/v2/config -H "Authorization: Bearer $token"
+
+# To connect using Nessie CLI:
+# connect to http://127.0.0.1:19120/api/v2 using
+#   "nessie.authentication.type"="OAUTH2" AND
+#   "nessie.authentication.oauth2.client-id"=client1 AND
+#   "nessie.authentication.oauth2.client-secret"=s3cr3t AND
+#   "nessie.authentication.oauth2.issuer-url"="http://127.0.0.1:8080/realms/iceberg/"
+
+
+version: '3'
+
+services:
+
+  ## IMPORTANT
+  ##
+  ## 1. Start the pods:
+  ##
+  ##      podman compose -f docker/catalog-auth-s3-otel/docker-compose.yml up
+  ##
+  ## 2. Once the Nessie Catalog pod is up, you can run any of the following commands to start a Spark SQL session with Nessie enabled:
+  ##    The script will ask for the client secret, it is 's3cr3t' for all three clients.
+  ##
+  ##      catalog/bin/spark-sql.sh --no-nessie-start --aws --oauth --client-id client1
+  ##      catalog/bin/spark-sql.sh --no-nessie-start --aws --oauth --client-id client2
+  ##      catalog/bin/spark-sql.sh --no-nessie-start --aws --oauth --client-id client3
+  ##
+  ##    Run the "standard smoketest SQL":
+  ##
+  ##      CREATE NAMESPACE nessie.sales;
+  ##      CREATE NAMESPACE nessie.eng;
+  ##
+  ##      USE nessie.sales;
+  ##      CREATE TABLE city (C_CITYKEY BIGINT, C_NAME STRING, N_NATIONKEY BIGINT, C_COMMENT STRING) USING iceberg PARTITIONED BY (bucket(16, N_NATIONKEY));
+  ##      INSERT INTO city VALUES (1, 'a', 1, 'comment');
+  ##      SELECT * FROM city;
+  ##
+  ##      USE nessie.eng;
+  ##      CREATE TABLE city (C_CITYKEY BIGINT, C_NAME STRING, N_NATIONKEY BIGINT, C_COMMENT STRING) USING iceberg PARTITIONED BY (bucket(16, N_NATIONKEY));
+  ##      INSERT INTO city VALUES (2, 'b', 2, 'comment');
+  ##      SELECT * FROM city;
+  ##
+  nessie:
+    image: ghcr.io/projectnessie/nessie:0.91.3
```

Comment:
This can be changed in the other compose files as well, now that we have it released.

Reply:
#9070